api/src/database/run-ast/utils/merge-with-parent-items.ts

```diff
@@ -46,10 +58,20 @@ export function mergeWithParentItems(
 
 		const parentsByPrimaryKey = new Map<PrimaryKey, Item>();
 
-		parentItems.forEach((parentItem: Item) => {
-			if (!parentItem[parentRelationField]) parentItem[parentRelationField] = [];
+		for (const parentItem of parentItems) {
+			if (!parentItem[parentRelationField]) {
+				parentItem[parentRelationField] = [];
+			}
+
+			let parentPrimaryKey: PrimaryKey | null = parentItem[parentPrimaryKeyField];
 
-			const parentPrimaryKey = parentItem[parentPrimaryKeyField];
+			// null if the user has no access to parent PK
+			if (parentPrimaryKey === null) {
+				continue;
+			} else {
+				// ensure key access is type agnostic
+				parentPrimaryKey = parentPrimaryKey.toString();
```

Comment:
Did you observe any specific bug that warrants this conversion?

We are not doing it in the M2O case above either and I would expect the fields to have the correct type when coming back from the DB?

Reply:
Yes this was the cause of the `O2M` error reported in the #24257

Reply:
Can confirm that this fixes the mismatched types when a BigInt field is used to store the Int PK and vice versa as the BigInt is stringified.